lib/msal-common/src/authority/Authority.ts

```diff
@@ -356,7 +356,8 @@ export class Authority {
         const canonicalAuthorityHost = this.hostnameAndPort;
         if (
             this.authorityType === AuthorityType.Adfs ||
-            !this.isAliasOfKnownMicrosoftAuthority(canonicalAuthorityHost)
+            (this.protocolMode !== ProtocolMode.AAD &&
```

Comment:
Note: This is an immediate solution, but it may be a good idea to add a way to either detect B2C authorities or indicate the authority is B2C in some way so B2C can be used in OIDC mode.

Reply:
We are not detecting CIAM so detecting B2C should be okay. The decision not to was made when we wanted to be IDP agnostic which is no longer true in many cases.